nerdctl: 0.5.0 -> 0.6.0

06kellyjac opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Bump nerdctl to `0.6.0`

Also:

- nixpkgs-fmt
- pull out go path
- add installCheckPhase
- hardcode homepage
- add changelog
- sort meta



###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

06kellyjac commented:
Result of `nixpkgs-review pr 113673` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>nerdctl</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/applications/networking/cluster/nerdctl/default.nix:
```diff
@@ -27,25 +22,43 @@ buildGoModule rec {
+    ];
+
+  postInstall =
+    let
+      binPath = lib.makeBinPath ([ buildkit ] ++ extraPackages);
+    in
+    ''
+      wrapProgram $out/bin/nerdctl \
+        --prefix PATH : "${binPath}" \
+        --prefix CNI_PATH : "${cni-plugins}/bin"
+    '';
```
```suggestion
    ''
      wrapProgram $out/bin/nerdctl \
        --prefix PATH : "${lib.makeBinPath ([ buildkit ] ++ extraPackages)}" \
        --prefix CNI_PATH : "${cni-plugins}/bin"
    '';
```

06kellyjac commented:
Result of `nixpkgs-review pr 113673` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nerdctl</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/applications/networking/cluster/nerdctl/default.nix:
```diff
@@ -27,25 +22,39 @@ buildGoModule rec {
-    "-X github.com/AkihiroSuda/nerdctl/pkg/version.Revision=<unknown>"
-  ];
+  buildFlagsArray =
+    let
+      t = "github.com/AkihiroSuda/nerdctl/pkg/version";
+    in
+    [
+      "-ldflags="
+      "-w"
+      "-s"
+      "-X ${t}.Version=v${version}"
+      "-X ${t}.Revision=<unknown>"
+    ];
```
```suggestion
    [
      "-ldflags="
      "-w"
      "-s"
      "-X github.com/AkihiroSuda/nerdctl/pkg/version.Version=v${version}"
      "-X github.com/AkihiroSuda/nerdctl/pkg/version.Revision=<unknown>"
    ];
```

06kellyjac replied:
this t = "gopath" shortening is pretty standard across nixpkgs. Either way reverted

06kellyjac committed:
nerdctl: 0.5.0 -> 0.6.0

- nixpkgs-fmt
- add installCheckPhase
- hardcode homepage
- add changelog
- sort meta

06kellyjac commented:
Result of `nixpkgs-review pr 113673` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nerdctl</li>
  </ul>
</details>

r-rmcgibbo commented:
Result of `nixpkgs-review pr 113673` at 1f1f14f5 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/86b293ca1e2165dfefe064dfa6811a2c">nerdctl</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>error: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-flags-array.md">no-flags-array</a>

`buildFlagsArray` is only intended to be used in Bash, not as a Nix attribute.

Near pkgs/applications/networking/cluster/nerdctl/default.nix:25:3:
```
   |
25 |   buildFlagsArray = [
   |   ^
```
</li>
  </ul>
</details>

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113673` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>nerdctl</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
nerdctl:
<details>

error: no-flags-array
`buildFlagsArray` is only intended to be used in Bash, not as a Nix attribute.

Near pkgs/applications/networking/cluster/nerdctl/default.nix:25:3:

```
   |
25 |   buildFlagsArray = [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-flags-array.md
</details>
</details>
